### PR TITLE
swap toml with tomli

### PR DESCRIPTION
**Pull request by graingert**
pip is moving from toml to tomli https://github.com/pypa/pip/pull/10035 towncrier should consider following suit

* pip: https://github.com/pypa/pip/pull/10035
* build: https://github.com/pypa/build/issues/308
* black: https://github.com/psf/black/pull/2301
* setuptools_scm https://github.com/pypa/setuptools_scm/pull/586

### CODE REVIEW

File: src/towncrier/_settings.py [hunk 1]
```diff
@@ -69,8 +69,8 @@ def load_config(directory):
 
 def load_config_from_file(directory, config_file):
 
-    with open(config_file, "r") as conffile:
-        config = toml.load(conffile)
+    with open(config_file, "r", encoding="utf8") as conffile:
```
Comment (graingert) [inline src/towncrier/_settings.py:72]: >  A TOML file must be a valid UTF-8 encoded Unicode document.

**Comment by twm** [comment]
It looks like you'll need to drop Python 3.5 support too.

**Comment by altendky** [comment]
What's the benefit of this switch to justify dropping <3.6?  FWIW, here's my take on supporting old versions.  https://github.com/pytest-dev/pytest-twisted/tree/5dc4efc5d335da0172fec02e48076aacef4bf75d#python-2-support-plans

**Comment by graingert** [comment]
> What's the benefit of this switch to justify dropping <3.6? FWIW, here's my take on supporting old versions. https://github.com/pytest-dev/pytest-twisted/tree/5dc4efc5d335da0172fec02e48076aacef4bf75d#python-2-support-plans

toml doesn't support toml v1 syntax

**Comment by hukkin** [comment]
As graingert mentioned, this will avoid issues like [this Black issue](https://github.com/psf/black/issues/2280), i.e. crashing hard if another tool uses TOML v1 syntax in the shared pyproject.toml.

Note that by setting `python_requires='>=3.6'` metadata in setup.py (currently not set), many legacy Python users will not notice Python 2 being dropped, as pip is smart enough to not install unsupported versions.

File: setup.py [hunk 2]
```diff
@@ -41,8 +40,9 @@
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
     use_incremental=True,
-    install_requires=["click", "click-default-group", "incremental", "jinja2", "setuptools", "toml"],
+    install_requires=["click", "click-default-group", "incremental", "jinja2", "setuptools", "tomli"],
     extras_require={"dev": ["packaging"]},
+    requires_python=">=3.5",
```
Comment (hukkin) [inline setup.py:45]:
```suggestion
    python_requires=">=3.6",
```

**Comment by altendky** [comment]
Well, yeah, that'll be a relevant reason...  `:|`  Thanks for the explanation.  After getting the last release out I've more or less disappeared so I've got no real say in process here at the moment, but is there any chance we could get another release out with whatever bugfixes before dropping 2.7 etc?

**Comment by graingert** [comment]
> Well, yeah, that'll be a relevant reason... `:|` Thanks for the explanation. After getting the last release out I've more or less disappeared so I've got no real say in process here at the moment, but is there any chance we could get another release out with whatever bugfixes before dropping 2.7 etc?

Yeah would be good to get another release

**Comment by graingert** [comment]
Could make this a conditional dep and drag 2.7 support on for a little while longer

File: setup.py [hunk 3]
```diff
@@ -41,7 +41,15 @@
...
     use_incremental=True,
-    install_requires=["click", "click-default-group", "incremental", "jinja2", "setuptools", "toml"],
+    install_requires=[
+        "click",
+        "click-default-group",
+        "incremental",
+        "jinja2",
+        "setuptools",
+        "toml; python_version < '3.6'",
+        "tomli; python_version > '3.5'",
```
Comment (graingert) [inline setup.py:51]:
```suggestion
        "tomli; python_version > '3.6'",
```

File: setup.py [hunk 4]
```diff
@@ -41,7 +41,15 @@
...
     use_incremental=True,
-    install_requires=["click", "click-default-group", "incremental", "jinja2", "setuptools", "toml"],
+    install_requires=[
+        "click",
+        "click-default-group",
+        "incremental",
+        "jinja2",
+        "setuptools",
+        "toml; python_version < '3.6'",
+        "tomli; python_version > '3.6'",
```
Comment (graingert) [inline setup.py:51]:
```suggestion
        "tomli; python_version >= '3.6'",
```

File: src/towncrier/_settings.py [hunk 5]
```diff
@@ -1,9 +1,15 @@
 # Copyright (c) Amber Brown, 2015
 # See LICENSE for details.
 
+import io
 import os
+import sys
 import pkg_resources
-import toml
+
+if sys.version_info > (3, 5):
```
Comment (graingert) [inline src/towncrier/_settings.py:9]:
```suggestion
if sys.version_info >= (3, 6):
```

File: src/towncrier/_settings.py (hunk 1 above)
Comment (graingert) [reply src/towncrier/_settings.py:78]:
```suggestion
    with io.open(config_file, "r", encoding="utf8") as conffile:
```

**Comment by graingert** [comment]
@altendky the conditional import isn't so bad py2.7 can survive another day

**Review by altendky — APPROVED** [review]
I appreciate the accommodation.

**Pull request merged**